.github/workflows/pr.yml

```diff
@@ -8,12 +8,11 @@ on:
       - main
   pull_request:
 jobs:
-  golangci-lint:
+  code-health:
```

Comment:
simplifying and joining linter and tests in the same job as their initialization is the same and it's what takes most of the time

Reply:
I think someone had strong opinion about small jobs but forgot who had that :D
I'm ok with larger PR job.

Reply:
i agree with that general idea, but in this case they have exactly the same initialization and second one depends on first one, so I thought they could be together, most of the code here is for initiazalition (which is duplicated in each former job)

Reply:
but happy to revert if you prefer that I leave them as they were

Reply:
I'm ok with the larger job